### PR TITLE
Add Sunglow plant mutation

### PR DESCRIPTION
**Pull request by drakewill-CRL**
## About the PR

Add a new plant mutation: Sunglow. It's Bioluminescent with the numbers cranked up to 8. The super-bright produce make a great light source in the event that the entire station loses power, as long as no mice sneak around to eat them.

## Why / Balance

Showing off that botany mutation effects can be added now without messing with the rest of botany code. Also allowing glowing plants to be of significantly more use on rare occasion.

I intend for this to be part of a 2nd process for rarer mutations (see [my Hydroponics proposal writeup](https://github.com/space-wizards/docs/pull/290) for a full explanation of that. For now, I want to start adding that content in small pieces and make it show up very rarely, so that when the 2nd process gets implemented in people do not feel like stuff is being taken away.

Odds: On a station with 3 full ChemMasters that get used purely for Unstable Mutagen, you can make 1,800u of mutagen. If the station uses all 1,800u in an hour-long shift, every shift, you will see this mutation approximately twice in a week at the current 1/100,000 odds. It would be once every 3 weeks at 1 in a million odds with the same scenario. I look forward to anecdotes about this showing up multiple times a round every round.

## Technical details

A new mutation is applied with a 1/100,000 chance. 
The Glow EntityEffect had a parameter added to enable this to work.  Beyond that, this is all YML. 

## Media

Standard Bioluminescent (for comparison):
image
Sunglow:
image

## Requirements

- [X] I have read and am following the [Pull Request and Changelog Guidelines](https://docs.spacestation14.com/en/general-development/codebase-info/pull-request-guidelines.html).
- [X] I have added media to this PR or it does not require an ingame showcase.

## Breaking changes

**Changelog**

:cl:
- add: Added a very rare plant mutation: Sunglow. Plants and produce may shine bright enough to illuminate rooms.

### CODE REVIEW

**Comment by ScarKy0** [comment]
Seeing how stacked glowing plants already can make areas so hellishly bright, will this one be even worse 🤔 
Also if its the sun it should also generate heat (imagine)

**Comment by drakewill-CRL** [comment]
> Seeing how stacked glowing plants already can make areas so hellishly bright, will this one be even worse 🤔 
This one isn't going to show up nearly as often, and the value in it is going to be to disperse them. Bioluminescent isn't particularly helpful so nobody bothers to keep a pocket flashlight-tomato instead of their PDA light.

> Also if its the sun it should also generate heat (imagine)
This is doable now that mutations are EntityEffects, but I don't think it's a fun idea to have produce cook you alive for having the audacity to see in a dark hallway.

**Comment by Ilya246** [comment]
to counter stacking it could produce a little bit of heat and perhaps combust if the air is too hot

**Comment by ScarKy0** [comment]
> This is doable now that mutations are EntityEffects, but I don't think it's a fun idea to have produce cook you alive for having the audacity to see in a dark hallway.

Well not THAT much heat, just that if for example you decide to stack them a lot you might need to permanently avoid botany from then on :godo:

**Comment by Ilya246** [comment]
combusting from too much temperature also introduces a natural failsafe so they don't heat the air past a certain point

**Comment by EthanQix** [comment]
> > This is doable now that mutations are EntityEffects, but I don't think it's a fun idea to have produce cook you alive for having the audacity to see in a dark hallway.
> 
> Well not THAT much heat, just that if for example you decide to stack them a lot you might need to permanently avoid botany from then on :godo:
> 
> Single instances wouldnt make a differance with the heat, i mean

Atmos usually can't keep the station's temperature in the double digits anyway, a heat producing plant would be the opposite of a problem for botanists.

As a side note, in order to not kill the lighting engine, this is a mutation that should affect only the plant but not the produce.

**Comment by WhiteGorila** [comment]
You are my sunshine, my only sunshine

File: Resources/Prototypes/Hydroponics/randomMutations.yml [hunk 1]
```diff
@@ -176,4 +176,10 @@
     - name: ChangeHarvest
       baseOdds: 0.036
       persists: false
-      effect: !type:PlantMutateHarvest
\ No newline at end of file
+      effect: !type:PlantMutateHarvest
+    - name: Sunglow
+      baseOdds: 0.000001 #1 in 100k units of mutagen
```
Comment (Partmedia) [inline Resources/Prototypes/Hydroponics/randomMutations.yml:181]:
```suggestion
      baseOdds: 0.000001 # 1 in 100k units of mutagen
```

**Review by Partmedia — COMMENTED** [review]
Thanks for the PR. This seems reasonable to me. Raising the radius/energy on a single pointlight won't break the engine, like some are claiming. The rarity is probably enough such that this won't be as game-breaking as some are worried about. I've kicked off a maintainer discussion thread for this.

**Comment by slarticodefast** [comment]
Hey, I gave it an in-game test and while this may be not a problem performance-wise this still causes significant visual problems with the light engine. These screenshots are with 5 and 10 sunglow cocoa beans respectively. And once you get the mutation, botany can easily produce hundreds of these in a short amount of time.
Screenshot (299)

Screenshot (300)

**Comment by ScarKy0** [comment]
>Hey, I gave it an in-game test and while this may be not a problem performance-wise this still causes significant visual problems with the light engine. These screenshots are with 5 and 10 sunglow cocoa beans respectively. And once you get the mutation, botany can easily produce hundreds of these in a short amount of time.

Ok I know this is supposed to be like bad but id absolutely love to walk past botany and get murdered by the power of the sun

**Comment by drakewill-CRL** [comment]
> this still causes significant visual problems with the light engine

My understanding is that this is how it's supposed to behave with a ton of intense lights in one spot. I agree that its' not very easy for players to see what's going on at the center, and I would hope that encourages players to spread these out in a useful way. There will be players looking to cause out-of-character problems with them, and I hate having to do primary design around that group, but I think most of that gets answered in the other discussion below.

>  And once you get the mutation, botany can easily produce hundreds of these in a short amount of time. 

I will open up a branch/PR for discussion shortly on "botany makes too much stuff and can cause problems". I have 2 or 3 ideas on possible answers, but I'd prefer to make that its own discussion and solution. This should just be for the Sunglow addition and its mertis or lack thereof, not botany yield debates.

**Comment by PJB3005** [comment]
Too many light sources are already known to break the engine outright, and even if they didn't they would be massive performance hogs and cause issues for people with weaker systems.

Without a completely different approach to light rendering for these, there is no way this feature can work from a technical perspective.

Giving them a one-in-a-million chance is not going to rectify that.

**Pull request closed without merging**